sky/resources.py

```diff
@@ -932,6 +946,29 @@ def _try_validate_ports(self) -> None:
         # We don't need to check the ports format since we already done it
         # in resources_utils.simplify_ports
 
+    def _try_validate_labels(self) -> None:
+        """Try to validate the labels attribute.
+
+        Raises:
+            ValueError: if the attribute is invalid.
+        """
+        if self._labels is None:
+            return
+
+        if self.cloud is None:
+            # Because each cloud has its own label format, we cannot validate
+            # the labels without knowing the cloud.
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError(
+                    'Cloud must be specified when labels are provided.')
+
+        # Check if the label key value pairs are valid.
+        for key, value in self._labels.items():
+            valid, err_msg = self.cloud.is_label_valid(key, value)
+            if not valid:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(f'Invalid label {key}={value}. {err_msg}')
```

Comment:
nit: it would be good to list all invalid labels all at once. Otherwise, the user will have to do multi-round of fixing. : )

Reply:
Good idea - updated with prettyTable now:
```
ValueError: The following labels are invalid:
        Label              Reason                                                                                                                                                                                                   
        cl/oud_name: gcp   Invalid label key cl/oud_name for GCP. Key must start with a lowercase letter and can include lowercase alphanumeric characters, dashes, and underscores, with a total length of 63 characters or less.  
        mylabel: my/value  Invalid label value my/value for GCP. Value can include lowercase alphanumeric characters, dashes, and underscores, with a total length of 63 characters or less.   
```